Tests/OpenCombineTests/ObservableObjectTests.swift

```diff
@@ -0,0 +1,432 @@
+
+        var counter = 0
+
+        observableObject.objectWillChange.sink {
+            counter += 1
+        }.store(in: &disposeBag)
+
+        XCTAssertEqual(observableObject.publishedValue0, 0)
+        XCTAssertEqual(observableObject.simpleValue, "what")
+        XCTAssertEqual(observableObject.subclassPublished0, 0)
+        XCTAssertEqual(observableObject.subclassPublished1, 1)
+        XCTAssertEqual(observableObject.subclassPublished2, 2)
+
+        observableObject.publishedValue0 += 5
+
+        XCTAssertEqual(counter, 1)
+        XCTAssertEqual(observableObject.publishedValue0, 5)
+
+        Published<String>[_enclosingInstance: observableObject,
+                          wrapped: \.simpleValue,
+                          storage: \.publishedValue1] += "???"
+
+        XCTAssertEqual(counter, 2)
+        XCTAssertEqual(observableObject.simpleValue, "what")
+
+        observableObject.subclassPublished0 += 3
+
+        XCTAssertEqual(counter, 3)
+        XCTAssertEqual(observableObject.subclassPublished0, 3)
+
+        observableObject.subclassPublished1 += 3
+
+        XCTAssertEqual(counter, 4)
+        XCTAssertEqual(observableObject.subclassPublished1, 4)
+
+        observableObject.subclassPublished2 += 3
+
+        XCTAssertEqual(counter, 5)
+        XCTAssertEqual(observableObject.subclassPublished1, 4)
+    }
+
+    func testObjCClassSubclass() {
+        let observableObject = ObjCClassSubclass()
+        let publisher1 = observableObject.objectWillChange
+        let publisher2 = observableObject.objectWillChange
+        XCTAssert(publisher1 === publisher2)
+    }
+
+    func testResilientClassSubclass() {
+        let observableObject = ResilientClassSubclass()
+        let publisher1 = observableObject.objectWillChange
+        let publisher2 = observableObject.objectWillChange
+
+        XCTAssert(publisher1 === publisher2)
+    }
+
+    func testResilientClassSubclass2() {
+        let observableObject = ResilientClassSubclass2()
+        let publisher1 = observableObject.objectWillChange
+        let publisher2 = observableObject.objectWillChange
+
+        XCTAssert(publisher1 === publisher2)
+    }
+
+    func testGenericClass() {
+        let observableObject = GenericClass(123, true)
+
+        var counter = 0
+
+        observableObject.objectWillChange.sink { counter += 1 }.store(in: &disposeBag)
+        XCTAssertEqual(counter, 0)
+        XCTAssertEqual(observableObject.value1, 123)
+        XCTAssertEqual(observableObject.value2, true)
+
+        observableObject.value1 += 1
+
+        XCTAssertEqual(counter, 1)
+        XCTAssertEqual(observableObject.value1, 124)
+
+        observableObject.value2.toggle()
+
+        XCTAssertEqual(counter, 2)
+        XCTAssertEqual(observableObject.value2, false)
+    }
+
+    // A bug in Combine (FB7471594). This has been fixed on iOS.
+    // But not deployed on other Darwin OS yet
+    #if !OPENCOMBINE_COMPATIBILITY_TEST || os(iOS)
```

Comment:
No, this condition is not needed. What I mean is that one should make sure to only run compatibility tests _against the most recent version_ of Combine (since that's what we aim to be compatible with). It so happens that the most recent version of Combine is usually vendored in the newest iOS version (not even the newest macOS version, and certainly not Catalina).

Tests should not be disabled based on that.

Just FYI, you can run the compatibility tests in the iOS simulator, this way they should pass. This is exactly what we do on our CI build.

Reply:
done